evidence/verify/verifiers/evidence_parser.go

```diff
@@ -0,0 +1,73 @@
+package verifiers
+
+import (
+	"encoding/json"
+	"io"
+
+	"github.com/jfrog/jfrog-cli-artifactory/evidence/dsse"
+	"github.com/jfrog/jfrog-cli-artifactory/evidence/model"
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	"github.com/pkg/errors"
+	"github.com/sigstore/sigstore-go/pkg/bundle"
+)
+
+// EvidenceParserInterface defines the interface for parsing evidence
+type EvidenceParserInterface interface {
+	ParseEvidence(evidence *model.SearchEvidenceEdge, evidenceResult *model.EvidenceVerification) error
+}
+
+type evidenceParser struct {
+	artifactoryClient artifactory.ArtifactoryServicesManager
+}
+
+func NewEvidenceParser(client *artifactory.ArtifactoryServicesManager) EvidenceParserInterface {
+	return &evidenceParser{
+		artifactoryClient: *client,
+	}
+}
+
+func (p *evidenceParser) ParseEvidence(evidence *model.SearchEvidenceEdge, evidenceResult *model.EvidenceVerification) error {
+	file, err := p.artifactoryClient.ReadRemoteFile(evidence.Node.DownloadPath)
+	if err != nil {
+		return errors.Wrap(err, "failed to read remote file")
+	}
+	defer func(file io.ReadCloser) {
+		_ = file.Close()
+	}(file)
+
+	fileContent, err := io.ReadAll(file)
+	if err != nil {
+		return errors.Wrap(err, "failed to read file content")
+	}
+	// Try Sigstore bundle first
+	if err := p.tryParseSigstoreBundle(fileContent, evidenceResult); err == nil {
+		return nil
+	}
+
+	// Fall back to DSSE envelope
+	if err := p.tryParseDsseEnvelope(fileContent, evidenceResult); err == nil {
+		return nil
+	}
+
+	return errors.New("failed to parse evidence as either Sigstore bundle or DSSE envelope")
```

Comment:
We're not returning what happened in case both sigstore parsing and dsse parsing failed the message is too generic. 
I suggest having different variable than err for each, sigstore and dsse, and return the error here.